server/types/api.go

```diff
@@ -23,24 +23,16 @@ type API interface {
 type APIOptions interface {
 	// Config returns the config data. Read only! Unsafe to edit.
 	Config() *config.Config
-
-	// Datastore holds access to local datastore.
-	// Used as a local data source to serve routing, storage, and caching.
-	Datastore() Datastore
```

Comment:
i should have probably handled it directly through this interface, but it required further extensions and handling different providers for storage and routing.

id like to bring it back here somehow, but it requires a bit more thought on how to handle provider details.